parse.go

```diff
@@ -93,6 +103,12 @@ type ParseOptions struct {
 
 	// If true, checks that directives have a valid number of arguments.
 	SkipDirectiveArgsCheck bool
+
+	// MatchFuncs are called in order when an unknown or non-core NGINX directive is
+	// encountered by the parser to determine the valid contexts and argument count of the
+	// directive. Set this option to enable parsing of directives belonging to non-core or
+	// dynamic NGINX modules that follow the usual grammar rules of an NGINX configuration.
+	MatchFuncs []MatchFunc
```

Comment:
might be better in term of API to only require one instead of a list? caller can do the "list" in their own implementation. Or something more optimal like a map

Reply:
Map felt worse for using the same value for multiple directives. The reason for the list is to allow the caller to specify multiple functions they may not have an ability to modify without additional boilerplate.

Reply:
discussed offline, possible use case for multiple is to make it easier for existing workflow of providing callback functions for different NAP.

Reply:
I expect as we want to add in directives from other dynamic modules that do not require special syntax we would add additional "match functions" instead of adding them to the big dictionary of directives. You probably wouldn't enable both versions of NAP at th the same time to parse a single config (although I guess you could?).